spaceship/lib/spaceship/connect_api/provisioning/client.rb

```diff
@@ -0,0 +1,57 @@
+require_relative '../client'
+require_relative '../../portal/portal_client'
+
+module Spaceship
+  class ConnectAPI
+    module Provisioning
+      class Client < Spaceship::ConnectAPI::Client
+        def self.instance
+          # Verify there is a token or a client that can be used
```

Comment:
I think we can remove a bunch of conditionals if we do something like:
`return @client if defined?(@client)` as the first line.

Reply:
@snatchev I don't think we can actually do that 🤔 These if statements that are here are used if a user wants to change users/teams during execution. `return @client if defined?(@client)` wouldn't reinitialize a new client when the a JWT or portal/tunes client is logged into

Reply:
Yes, you are 💯 correct. In my mind, I read the or conditional as a nil guard... which it is not!